packages/react-popup/Spec.md

```diff
@@ -2,30 +2,375 @@
+  calloutProps={{bounds, target}}
+/>
+```
+
+v8 `Callout` has no notion of separate boundaries for flip or overflow, and auto behaviour is used for flip and overflow 'pushing'.
+
+### Events
+
+v8 provides the following positioning event callbacks
+
+- onLayerMounted -> proposed to be removed in converged Portal spec [#17824](https://github.com/microsoft/fluentui/pull/17824)
+- onPositioned
+- onScroll
+
+### Hidden mount
+
+v8 `Callout` provides two props which will allow mounting a hidden popup and disabling renders when the component is hidden. According to the PRs that introduced the features, it should be a performance optimization.
+
+- `hidden` [#4419](https://github.com/microsoft/fluentui/pull/4419)
+- `shouldUpdateWhenHidden` [#10465](https://github.com/microsoft/fluentui/pull/10465)
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<Popup>
+  <PopupTrigger>
+    <button>Opens popup</button>
+  </PopupTrigger>
+
+  <PopupContent>
+    <h1>Popup</h2>
+    <div>Some section</div>
+    <div>Some section</div>
+  </PopupContent>
+</Popup>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+- Anchor content to a different target than the trigger
+- Inline popup rendered in DOM order
+- Open on:
+  - Click
+  - Hover
+  - Context menu (right click)
+  - Focus
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+The `Popup` component will use React context to manage both a trigger and content component.
+
+### Popup
+
+Outer component that setups context and does not render DOM.
+
+> TODO Discuss: dismiss on scroll ?
+
+> TODO Discuss: v8 `hidden` `shouldUpdateHidden` prop [#4419](https://github.com/microsoft/fluentui/pull/4419) [#10465](https://github.com/microsoft/fluentui/pull/10465)
+
+> TODO Discuss: v8 `onPositioned`
+
+> TODO Discuss: v0/v8 `autoFocus/setInitialFocus` ?
+
+> TODO Discuss: start small with API base for positioning props ?
+
+The `@fluentui/react-positioning` library that exports the `usePopper` hook which will power the `Popup` contains more than the declared props here. These extra positioning props should be exposed as required.
+
+```typescript
+export interface PopupProps {
+  /**
+   * Explicitly render the popup in DOM order
+   */
+  inline?: boolean;
+
+  /**
+   * Controls the popup open state
+   */
+  open?: boolean;
+
+  /**
+   * Call back when the component requests to change value
+   */
+  onOpenChange?: (e: OpenPopupEvents, data: OpenEventData) => void;
+
+  /**
+   * Anchor the popup to an element other than the trigger
+   */
+  target?: HTMLElement;
+
+  /**
+   * Popup position relative to target
+   */
+  position?: string;
+
+  /**
+   * Popup alignment relative to target
+   */
+  align?: string;
```

Comment:
Can discuss also this? Usage of `align` & `position` is non-intuitive, in N* we had few offline discussions to merge them into a single prop.